src/rgw/rgw_rest_s3.cc

```diff
@@ -6391,6 +6453,69 @@ int RGWSelectObj_ObjStore_S3::extract_by_tag(std::string tag_name, std::string&
   return 0;
 }
 
+size_t RGWSelectObj_ObjStore_S3::get_obj_size()
+{
+  return s->obj_size;
+}
+
+int RGWSelectObj_ObjStore_S3::range_request(int64_t ofs, int64_t len,void *buff,optional_yield* y)
+{
+  //purpose: implementation for arrow::ReadAt, this may take several async calls.
+  //send_response_date(call_back) accumulate buffer, upon completion it signal future.
+  range_req_str = "bytes=" + std::to_string(ofs) + "-" + std::to_string(ofs+len-1);
+  range_str = range_req_str.c_str();
+  range_parsed = false;//TODO to remove; 
+  RGWGetObj::parse_range();
+  requested_buffer.clear();
+  m_request_range = len;
+
+  ldout(s->cct, 10) << "S3select: calling execute(async):" << " offset :" << ofs << " length :" << len << "buffer size : " << requested_buffer.size() << dendl;
+
+  range_request_complete = std::promise<void>();
+  range_request_complete_future = range_request_complete.get_future();
+  RGWGetObj::execute(*y);
+
+  range_request_complete_future.wait(); //wait until whole range is retrieved //TODO timeout
+
```

Comment:
please stop with blank lines on every other line of code;  it doesn't help readability

Reply:
ok